src/lib/protocols/credentials/CredentialService.ts

```diff
@@ -266,6 +280,12 @@ export class CredentialService extends EventEmitter {
     return this.credentialRepository.find(id);
   }
 
+  private validateState(current: CredentialState, valid: CredentialState) {
```

Comment:
Maybe call it `assertState` instead? To me validate state suggests there is some custom validation done in the function, while it basically checks if the first state passed is equal to the second state passed.

```suggestion
  private assertState(current: CredentialState, expected: CredentialState) {
```

Reply:
That's good point, let's change it. Updated.

Reply:
I'm thinking that we could create another object, something like `CredentialStateMachine` which would be responsible for such assertions and state transitions. It could simplify tests. I haven't thought through all the details yet.